sponsorship/views.py

```diff
@@ -30,4 +32,20 @@ def sponsorship_delete(request, sponsorship_slug):
 
 def sponsorship_details(request, sponsorship_slug):
     sponsorship = get_object_or_404(Sponsorship, slug=sponsorship_slug)
-    return render(request, 'sponsorship/sponsorship_details.html', {'sponsorship': sponsorship})
\ No newline at end of file
+    return render(request, 'sponsorship/sponsorship_details.html', {'sponsorship': sponsorship})
+
+def sponsorship_edit(request, sponsorship_slug):
+    sponsorship_toupdate = get_object_or_404(Sponsorship, slug=sponsorship_slug)
+    if request.method == "POST":
+        form = create_sponsorship_form(request.POST, instance=sponsorship_toupdate)
+        if form.is_valid():
+            try:
+                form.save()
+                return redirect('sponsorship_list')
+            except ValidationErr as v:
+                messages.error(request, str(v.args[0]))
+        else:
+            messages.error(request, 'Formulario con errores')
+
+    form = create_sponsorship_form(instance=sponsorship_toupdate)
+    return render(request, 'sponsorship/sponsorship_form.html', {"form": form})
```

Comment:
When triying to update a sponsorship, this error appear:
![image](https://user-images.githubusercontent.com/80269612/227764675-cad48cdc-5b39-4897-b08a-a3bc216d9656.png)
It seems like it is triying to create another object sponsorship instead of updating it, therefore and slug duplication exception appear,